test/e2e/agent/agent_test.go

```diff
@@ -63,4 +64,317 @@ var _ = Describe("VM Agent behavior", func() {
 				enrollmentID).ShouldNot(BeNil())
 		})
 	})
+
+	Context("status", Label("75991"), func() {
+		It("should report the correct device status after an inline config is added", func() {
+			deviceId := harness.GetEnrollmentIDFromConsole()
+			logrus.Infof("Enrollment ID found in VM console output: %s", deviceId)
+			Expect(deviceId).NotTo(BeNil())
+
+			// Approve the enrollment and wait for the device details to be populated by the agent.
+			harness.ApproveEnrollment(deviceId, testutil.TestEnrollmentApproval())
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeEmpty())
+			logrus.Infof("The device %s was approved", deviceId)
+
+			// Wait for the device to pickup enrollment and report measurements on device status.
+			Eventually(harness.GetDeviceWithStatusSystem, TIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeNil())
+			logrus.Infof("The device %s is reporting its status", deviceId)
+
+			// Check the device status.
+			response := harness.GetDeviceWithStatusSystem(deviceId)
+			device := response.JSON200
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+			Expect(*device.Status.Summary.Info).To(Equal("Bootstrap complete"))
+			Expect(device.Status.Updated.Status).To(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				logrus.Infof("Current device is %s", deviceId)
+
+				// Create ConfigProviderSpec.
+				var configProviderSpec v1alpha1.ConfigProviderSpec
+				err := configProviderSpec.FromInlineConfigProviderSpec(validInlineConfig)
+				Expect(err).ToNot(HaveOccurred())
+
+				device.Spec.Config = &[]v1alpha1.ConfigProviderSpec{configProviderSpec}
+				logrus.Infof("Updating %s with config %s", deviceId, device.Spec.Config)
+			})
+
+			logrus.Infof("Waiting for the device to pick the config")
+			harness.WaitForDeviceContents(deviceId, "the device is upgrading to renderedVersion: 2",
+				func(device *v1alpha1.Device) bool {
+					for _, condition := range device.Status.Conditions {
+						if condition.Type == "Updating" && condition.Reason == "Updated" && condition.Status == "False" &&
+							condition.Message == "Updated to desired renderedVersion: 2" {
+							return true
+						}
+					}
+					return false
+				}, TIMEOUT)
+
+			// The device should have the online config.
+			stdout, err := harness.VM.RunSSH([]string{"cat", "/etc/motd"}, nil)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(stdout.String()).To(ContainSubstring("This system is managed by flightctl."))
+
+			// The status should be "Online"
+			logrus.Infof("The device has the config %s", device.Spec.Config)
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
```

Comment:
```suggestion
			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
```

do we really need LONGTIMEOUT in this case? I wonder if it's worth having something intermediate, i.e. if something is going wrong with a test, we detect it earlier.

I would hope that "Online" at this point is very short or nothing.

Reply:
Right, updated, thanks